streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -516,11 +528,97 @@ public void shouldNotAllowNullTopicChooserOnTo() {
         testStream.to((TopicNameExtractor<String, String>) null);
     }
 
-    @Test(expected = NullPointerException.class)
+    @Test
     public void shouldNotAllowNullTransformSupplierOnTransform() {
+        exception.expect(NullPointerException.class);
+        exception.expectMessage("transformerSupplier can't be null");
         testStream.transform(null);
     }
 
+    @Test
+    public void shouldReturnSingletonListOrEmptyListIfSinglePairTransformerReturnsNotNullOrNull() {
```

Comment:
Ack. Makes sense.

I am wondering, if we should extract the part that builds the wrapper into its own method (package private so we can test it) and simplify the test accordingly?

(Or at least adding a comment might help -- not sure if we can encode the purpose in the test name -- it's too complex IMHO.)

Thoughts?

Reply:
I extracted the wrapper code into an adapter class (`TransformSupplierAdapter`). It seemed to me to be the cleaner solution instead of loosening access to members for unit tests.